lib/web_ui/lib/src/engine/keyboard.dart

```diff
@@ -91,9 +91,9 @@ class Keyboard {
 
     final String timerKey = keyboardEvent.code;
 
-    // Don't synthesize a keyup event for modifier keys because the browser always
-    // sends a keyup event for those.
-    if (!_isModifierKey(event)) {
+    // Don't synthesize a keyup event for modifier keys, or keys not affected by modifiers,
+    // because the browser always sends a keyup event for those.
+    if (!_isModifierKey(event) && _isAffectedByModifiers(event)) {
```

Comment:
To fix the issue that's making tests fail, let's move the new condition inside. This way we guarantee that the syntehsizing timer is always canceled.

Does something like this work:
```dart
if (!_isModifierKey(event)) {
  _keydownTimers[timerKey]?.cancel();
  if (event.type == 'keydown' && _isAffectedByModifiers(event)) {
    _keydownTimers[timerKey] = Timer(_keydownCancelDuration, () {
      _keydownTimers.remove(timerKey);
      _synthesizeKeyup(event);
    });
  } else {
    _keydownTimers.remove(timerKey);
  }
}
```

Reply:
I remembering trying that, but that didn't solved the issue (flutter/flutter#58727), but right now, I can't remember why,  I will give this another try, because looking again, this suggestion should work! Maybe I have confused things.

Reply:
@erickzanardo did you have a chance to try it out?

Reply:
Hey, I haven't yet, sorry, I have been really overwhelmed at work. But I am planning on getting back to this PR on this saturday or sunday.

Reply:
No rush.

If the suggestion doesn't work, please ping so we can discuss it further.